backend/service/github/github.go

```diff
@@ -374,20 +396,26 @@ func (s *svc) CreateBranch(ctx context.Context, req *CreateBranchRequest) error
 		return err
 	}
 
-	for filename, contents := range req.Files {
+	for filePath, contents := range req.Files {
 		if contents == nil {
-			if err := wt.Filesystem.Remove(filename); err != nil {
-				return err
+			if err := wt.Filesystem.Remove(filePath); err != nil {
```

Comment:
So I don't think all this new logic is necessary. I should've seen it in the other PR but we can just rotate from `wt.Filesystem.Remove` to `wt.Remove` (https://github.com/go-git/go-git/blob/master/worktree_status.go#L573)

Which will delete it if it's a file or a directory